pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go

```diff
@@ -104,25 +113,42 @@ func (s *apiScaffolder) Scaffold() error {
 		return fmt.Errorf("error updating config/samples: %v", err)
 	}
 
-	controller := &controllers.Controller{
-		ControllerRuntimeVersion: golangv3scaffolds.ControllerRuntimeVersion,
-	}
-	if err := scaffold.Execute(
-		controller,
-	); err != nil {
-		return fmt.Errorf("error scaffolding controller: %v", err)
-	}
+	if isLegacyLayout {
+		controller := &controllers.Controller{
+			ControllerRuntimeVersion: golangv3scaffolds.ControllerRuntimeVersion,
+			IsLegacyLayout:           true,
+		}
```

Comment:
Maybe initialize the `Controller` struct with the isLegacy variable - to avoid duplicating the following code?

Reply:
We will set the defaults into `func (f *Controller) SetTemplateDefaults() error {` so, we need to have the value there.

Reply:
Note that the "duplication" code here is required because we are initializing different Controller{} implementations one is from v3 and another is from v4. So it is not duplicated actually.